darksky/request_manager.py

```diff
@@ -0,0 +1,38 @@
+import requests
+import aiohttp
+from .exceptions import DarkSkyException
+
+
+class BaseRequestManger(object):
+    def __init__(self, gzip: bool):
+        self.headers = {} if not gzip else {'Accept-Encoding': 'gzip'}
+
+    def make_request(self, url: str, **params):
+        raise NotImplementedError
+
+
+class RequestManger(BaseRequestManger):
+    def __init__(self, gzip: bool):
+        super().__init__(gzip)
+        self.session = requests.Session()
+        self.session.headers = self.headers
+
+    def make_request(self, url: str, **params):
+        response = self.session.get(url, params=params).json()
+        if 'error' in response:
+            raise DarkSkyException(response['code'], response['error'])
+        return response
+
+
+class RequestMangerAsync(BaseRequestManger):
```

Comment:
def __init__(self, client_session=None):
        # import here so aiohttp is not library dependency
        import aiohttp

        if client_session is None:
            self.session = aiohttp.ClientSession()
        else:
            assert isinstance(client_session, aiohttp.ClientSession)
            self.session = client_session

Reply:
Sure

Reply:
I'll create a PR in a sec

Reply:
@spacemanspiff2007 Is it a good idea to remove aiohttp from dependencies? We provide this functionality out of the box, so I think we need it in the setup.py file and import it on top

Reply:
It could be like this
```python
class RequestMangerAsync(BaseRequestManger):
    def __init__(self, gzip: bool):
        super().__init__(gzip)
        self.session = aiohttp.ClientSession(headers=self.headers)

    async def make_request(self, url: str, **params):
        # Fix for yarl(Doesn't support any types besides str)
        for key in params.copy():
            if params[key] is None:
                del params[key]

        response = await self.session.get(url, params=params, headers=self.headers)
        data = await response.json()
        if 'error' in data:
            raise DarkSkyException(data['code'], data['error'])
        return data```

But user need to close connection before exit

Reply:
However you like. aiohttp has quite a lot of dependencies and I think the majority of users uses requests anyway.
If your claim is that after installation everthing has to work out of the box then it must be added in setup.py. Otherwise a hint in the readme would be sufficient (imho).